### ISSUE TITLE
Gcloud component unexpectedly choosing beta component 

### ISSUE THREAD

**Issue by barm** [google-github-actions/deploy-cloudrun#376]
### TL;DR

I'm expecting that my deploy command will be run with the general availability component but it's actually using the beta component. The docs seem to indicate that it should use general availability by default, but it looks like there are a couple times in the source code where it does this: 

```
gcloudComponent = gcloudComponent ?? 'beta';
```

### Expected behavior

use general availability component

### Observed behavior

use beta component

### Action YAML

```yaml
staging-deploy:
    if: github.ref == 'refs/heads/main'
    runs-on: ubuntu-latest
    needs: build
    concurrency: staging-deploy
    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
      - name: Get Short SHA
        id: vars
        run: echo "::set-output name=sha_short::$(echo ${{ github.sha }} | cut -c1-10)"

      - name: Get Revision Tag
        id: tag
        run: | 
          echo "::set-output name=name::$(
            echo "t-${{ steps.vars.outputs.sha_short }}"
          )"

      - name: Authenticate with Staging GCP Service User
        uses: 'google-github-actions/auth@v0'
        with:
          project_id: ${{ env.GCP_PROJECT_ID_STAGING }}
          credentials_json: '${{ secrets.GCP_ACCOUNT_KEY }}'

      - name: Deploy to Cloud Run
        id: deploy
        uses: google-github-actions/deploy-cloudrun@v0
        with:
          service: ${{ env.GCP_CLOUD_RUN_SERVICE_STAGING }}
          image: ${{ env.ARTIFACT_PATH }}
          region: ${{ env.GCP_REGION }}
          no_traffic: true
          tag: ${{ steps.tag.outputs.name }}
          suffix: ${{ steps.vars.outputs.sha_short }}

      - name: Check Health
        uses: jtalk/url-health-check-action@v2
        with:
          url: "${{ steps.deploy.outputs.url }}/health"
          max-attempts: 10
          retry-delay: 500ms

      - name: Switchover Traffic
        run: |
          gcloud run services update-traffic ${{ env.GCP_CLOUD_RUN_SERVICE_STAGING }} \
            --region ${{ env.GCP_REGION }} \
            --to-revisions ${{env.GCP_CLOUD_RUN_SERVICE_STAGING}}-${{ steps.vars.outputs.sha_short }}=100 \
            --remove-tags ${{ steps.tag.outputs.name }}

      # Handle failures:
      - name: Remove Tag
        if: ${{ failure() }}
        run: |
          gcloud run services update-traffic ${{ env.GCP_CLOUD_RUN_SERVICE_STAGING }} \
            --region ${{ env.GCP_REGION }} \
            --remove-tags ${{ steps.tag.outputs.name }}

      - name: Trigger PagerDuty
        if: ${{ failure() }}
        run: |
          curl -X POST https://events.pagerduty.com/v2/enqueue \
            --fail \
            -H 'Content-Type: application/json' \
            -d '{ "routing_key": "${{ secrets.PAGERDUTY_INTEGRATION_KEY_STAGING }}", "event_action": "trigger", "dedup_key": "staging-deploy", "payload": {"summary": "staging deploy failed", "source": "ci", "severity": "info"} }'
```

### Log output

```text
Run google-github-actions/deploy-cloudrun@v0
  with:
    service: XXXX
    image: us-docker.pkg.dev/XXXX/XXXX/XXXX
    region: us-central1
    no_traffic: true
    tag: t-a24f98d5cf
    suffix: a24f98d5cf
  env:
    ARTIFACT_PATH: us-docker.pkg.dev/XXXX/XXXX/XXXX
    GCP_REGION: us-central1
    GCP_PROJECT_ID_INFRASTRUCTURE: XXXX
    GCP_PROJECT_ID_STAGING: XXXX
    GCP_CLOUD_RUN_SERVICE_STAGING: XXXX
    GCP_PROJECT_ID_PRODUCTION: XXXX
    GCP_CLOUD_RUN_SERVICE_PRODUCTION: XXXX
    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: XXXX
    GOOGLE_APPLICATION_CREDENTIALS: XXXX
    GOOGLE_GHA_CREDS_PATH: XXXX
    CLOUDSDK_CORE_PROJECT: XXXX
    CLOUDSDK_PROJECT: XXXX
    GCLOUD_PROJECT: XXXX
    GCP_PROJECT: XXXX
    GOOGLE_CLOUD_PROJECT: XXXX
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/c63efa65-2ba4-46ee-9997-978f7eb0f235 -f /home/runner/work/_temp/308d9b4e-0e44-4709-93f6-5767a9c5a42b
Running: gcloud beta run deploy {SERVICE_NAME} --image us-docker.pkg.dev/{PROJECT}/{ARTIFACT_REPO}/{ID} --quiet --platform managed --region us-central1 --tag t-a24f98d5cf --revision-suffix a24f98d5cf --no-traffic --project {PROJECT} --format json
```

### Additional information

If I could pass in a flag for the GA component that would also solve my issue but it will throw if the value is not alpha or beta.

**Comment by barm** [comment]
More context on this: I've seen a couple instances of unreliability with deployments failing through github actions but able to succeed through the web console. When I've made support cases, I've been told that I need to use the GA command instead: 

> Can you try using the general available command instead? You may refer to this documentation [2]. Please do let me know if this worked. This is the final requirement before we raise this to the Cloud Run Engineering team.

**Comment by sethvargo** [comment]
Hi @barm 

Certain CLI commands and flags are only available in the `beta` or `alpha` track. It's possible some flags have been promoted to GA, but that's why you're seeing this behavior.

**Comment by barm** [comment]
Thanks for the reply. Based on the docs it looks like the commands this projects supports are available in GA and Google support is not providing me support unless I use the GA commands.

https://cloud.google.com/sdk/gcloud/reference/run/deploy
https://cloud.google.com/sdk/gcloud/reference/run/services/update-traffic

**Comment by barm** [comment]
@sethvargo Much appreciated!

**Comment by sethvargo** [comment]
It will probably take a few days to get that merged and released. In the interim, are you able to use `setup-gcloud` directly and run the (GA) command? If the issue persists, Cloud Support should be able to unblock you then.

**Comment by barm** [comment]
Yeah I can definitely do that in the meantime and am actually already doing it for the update traffic command. Thanks again!

**Comment by barm** [comment]
Hey just checking in on this. Any ETA on the package bump? Thanks!

**Comment by barm** [comment]
Hi still waiting on this. We have the commit SHA pinned explicitly, but would be nice to be able to switch over to a release. Thanks!
